### DISCUSSION TITLE
Opening discussion from Old Forum regarding Java 17, Jakarta EE, latest PrimeFaces version and Gradle

### DISCUSSION THREAD

**Discussion by aleon1220** [primefaces/community#1234] [PrimeFaces] [answered]
# Introduction
Hello everyone, 
I would love to continue the conversation based on the old forum post [Primefaces+OpenJDK 17+wildfly:27.0.0.Final](https://forum.primefaces.org/viewtopic.php?f=3&t=79394)

# Context
I will shoot a couple of questions below in a numbered list below. First, I want to give you some context.

I have been experimenting and modernising a webapp i wrote in 2018 with PF 6 (I know is old). Below some key decisions part of my learning experiment.

- The webapp must use gradle with Kotlin syntax for flexibility. https://github.com/aleon1220/multi-cloud-WebApp-Attendance/blob/feature/app-beautification/build.gradle.kts
- The webapp will run as a docker-compose stack to have some other containers available for local testing e.g. OpenLDAP, MongoDB, etc
- The Webapp must use java 17 or any LTS version
- The final webapp must be functional and more up to 2024 java web development standard with PF
- Now it was running with Tomcat 9 and java 11. I am OK with any other suggestion for servlet container. I have heard of Wildfly, TomEE and Glassfish

# Questions
1. Is using Primefaces 14 with Jakarta EE latest OK?
2. Is using Java 17 LTS or higher recommendable?
3. Running the stack in a docker-compose container orchestration form. constructive thoughts?
4. What is recommended apart from Tomcat to run the webapps nowadays?

## References
some references I have found online that might help other users.

1. https://www.primefaces.org/documentation/
2. https://github.com/orgs/primefaces/discussions/1100
3. https://github.com/primefaces/primefaces/pull/7811
4. https://github.com/primefaces/primefaces/pull/7811
5. https://github.com/orgs/primefaces/discussions/43
6. https://github.com/orgs/primefaces/discussions/21
7. https://stackoverflow.com/questions/76330164/how-to-retrieve-facescontext-with-java-17-spring-6-jakarta-server-faces-4-x-an
8. github.com/primefaces/primefaces/issues/11433
9. https://github.com/primefaces/primefaces/issues/11434
10. https://discord.com/channels/787967399105134612/787967662293909524
11. https://stackoverflow.com/questions/tagged/primefaces+jsf
12. https://www.primefaces.org/showcase/index.xhtml?jfwid=1d2b6
13. https://www.primefaces.org/docs/api/6.0/
14. https://www.primefaces.org/issuetracker/
15. https://github.com/eclipse-ee4j/mojarra

**Comment by tandraschko** [comment, marked as answer]
1) yes, even 13.0 is 
2) yes
3) thats totally ok
4) there are different container. If you like docker and dont need session replication, you can try JSF on Quarkus. We use TomEE, however 10.0 isn't released yet. Its the smallest and fastest oldschool container.

**Reply by melloware** [reply]
Concur with everything @tandraschko has said but extra info for number 4.. I use Wildfly 31 which is Open Source, fast, and supports EE10 Jakarta.  But container will be your preference.

As for running in a container you might want to check out Quarkus: https://github.com/melloware/quarkus-faces

it has Docker builds out of the box so you can build right to your Docker Container.

**Reply by aleon1220** [reply]
thanks for your response. The quarkus-faces project is super interesting. I want to learn Quarkus and a bit more springBoot first. 

As for now I see that Wildfly is certified and quite popular so i will test my setup with Wildfly first https://jakarta.ee/compatibility/certification/10/

one of my biggest concerns is that majority prefers maven and I migrated from maven a few months ago because i did lots of android builds work and saw more flexibility with gradle.

Thanks everyone
God bless

**Comment by aleon1220** [comment]
> 1. yes, even 13.0 is
> 2. yes
> 3. thats totally ok
> 4. there are different container. If you like docker and dont need session replication, you can try JSF on Quarkus. We use TomEE, however 10.0 isn't released yet. Its the smallest and fastest oldschool container.

thank you. This helps a lot.
